packages/creator/src/lib/events/events.ts

```diff
@@ -27,23 +33,52 @@ export default class Events<Strategy extends TransactionStrategy> {
     params: CreateAndStoreEventParameters,
     overrides: NonPayableOverrides = {}
   ): Promise<CreatedEvent> {
-    await getCreatorIdentity(this.creator); // assert has identity
+    // assert that the creator has an identity to store the content if not using the smart asset issuer privacy gateway
+    if (!params.useSmartAssetIssuerPrivacyGateway) {
+      await getCreatorIdentity(this.creator);
```

Comment:
isn't it redundant since th ere is a getCreatorIdentity in storeEvent() ?

Reply:
yes right